app/controllers/schools_controller.rb

```diff
@@ -1,10 +1,23 @@
 # frozen_string_literal: true
 
 class SchoolsController < ApplicationController
+  layout 'info_site/application'
   
   def index
-    @schools = School.active
-
+    # Group by first letter of the school.
+    @schools = School.active.group_by { |i| i.name[0] }
+    @schools_arr = []
+    @schools.each do |key, value|
+      school_hash = {}
+      # If school name is start with alphabet letter, school names will display under aplhabet anchor eg: 'A' Academy for Careers (12), Academy(22).
+      # If school name is not start with alphabet letter, school names will display under '#' anchor. eg: '#' 486 newyork school (86),  56test(234).
+      school_hash['schoolname_first_letter'] = key.match?(/[A-Za-z]/) ? key : '#'
+      school_hash['school_names'] = value.collect do |i|
```

Comment:
I'd only use `i` for an integer variable. These are properties on schools, right? This logic looks a little odd to me. Couldn't we get the schools in turn and just do `school.code`? It's totally fine if that's not the best way to do it, but I think this approach needs a bit more commenting to make it clear to me. Thanks!